frontend/projects/ui/src/app/components/form-object/form-object.component.ts

```diff
@@ -62,31 +62,36 @@ export class FormObjectComponent {
   ) {}
 
   ngOnInit() {
-    Object.keys(this.objectSpec).forEach(key => {
-      const spec = this.objectSpec[key]
-
-      if (spec.type === 'list' && ['object', 'union'].includes(spec.subtype)) {
-        this.objectListDisplay[key] = []
-        this.formGroup.get(key)?.value.forEach((obj: any, index: number) => {
-          const displayAs = (spec.spec as ListValueSpecOf<'object'>)[
-            'display-as'
-          ]
-          this.objectListDisplay[key][index] = {
+    if (this.objectSpec) {
```

Comment:
Cleaner to do `Object.keys(this.objectSpec || {})` and remove higher conditional. Or another option would be `if (!this.objectSpec) return`